src/core/Core.js

```diff
@@ -35,12 +30,181 @@ export default class Core {
 
     this.translator = new Translator({locales: this.opts.locales})
     this.i18n = this.translator.translate.bind(this.translator)
-    // console.log(this.i18n('filesChosen', {smart_count: 3}))
 
     // Set up an event EventEmitter
     this.emitter = new ee.EventEmitter()
 
-    this.totalFilesSelectedCount = 0
+    this.defaultState = {
+      selectedFiles: {},
+      uploadedFiles: {},
+      modal: {
+        isHidden: true,
+        targets: []
+      }
+    }
+
+    this.state = Object.assign({}, this.state, this.defaultState)
+  }
+
+  /**
+   * Iterate on all plugins and run `update` on them. Called when state changes
+   *
+   */
+  updateAll () {
+    Object.keys(this.plugins).forEach((pluginType) => {
+      this.plugins[pluginType].forEach((plugin) => {
+        plugin.update(this.state)
+      })
+    })
+  }
+
+  /**
+   * Reset state to defaultState, used when Modal is closed, for example
+   *
+   */
+  resetState () {
+    this.setState(this.defaultState)
+  }
+
+  /**
+   * Updates state
+   *
+   * @param {newState} object
+   */
+  setState (newState) {
+    this.log(`Update state with: ${newState}`)
+    this.state = Object.assign({}, this.state, newState)
+    this.updateAll()
+  }
+
+  /**
+   * Gets current state, making sure to make a copy of the state object and pass that,
+   * instead of an actual reference to `this.state`
+   *
+   */
+  getState () {
+    return Object.assign({}, this.state)
```

Comment:
My first thought was someone may think it's the actual state, so they'd call `Object.assign` on it again.  There may also be an issue if someone fetched the state with `getState`, then waited for an async callback to fire before updating the state.  If something else altered the actual state before that callback is fired, there could be issues.  Idk if that situation would realistic arise, but I could see that being a race condition situation.